keras_retinanet/models/resnet.py

```diff
@@ -19,66 +19,69 @@
 import keras_resnet.models
 from ..models import retinanet
 
-WEIGHTS_PATH_NO_TOP_50 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-50-model.keras.h5'
-WEIGHTS_PATH_NO_TOP_101 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-101-model.keras.h5'
-WEIGHTS_PATH_NO_TOP_152 = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/ResNet-152-model.keras.h5'
+resnet_filename = 'ResNet-{}-model.keras.h5'
+resnet_resource = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/{}'.format(resnet_filename)
 
 custom_objects = retinanet.custom_objects.copy()
 custom_objects.update(keras_resnet.custom_objects)
 
 
-def ResNet50RetinaNet(inputs, num_classes, weights='imagenet', *args, **kwargs):
-    image = inputs
+def resnet_retinanet(num_classes, backbone='50', inputs=None, weights='imagenet', **kwargs):
+    allowed_backbones = ['50', '101', '152']
+    if backbone not in allowed_backbones:
+        raise ValueError('Backbone (\'{}\') not in allowed backbones ({}).'.format(backbone, allowed_backbones))
 
-    # load pretrained imagenet weights?
+    # choose default input
+    if inputs is None:
+        inputs = keras.layers.Input(shape=(None, None, 3))
+
+    # determine which weights to load
     if weights == 'imagenet':
```

Comment:
I suppose this isn't new behaviour, but this seems like a weird place to download the weights. It also means that you can't use a file called 'imagenet' as weights. Not a huge problem, but it is a bit dirty.

Personally I'd put the downloading in a separate function that a script can invoke before creating the model. That would leave this function a lot cleaner.

Reply:
Fair enough.